behavior.nls

```diff
@@ -329,87 +308,77 @@ to duplicate-remover
     ; by working further on the attacked argument, unless it discoveres a child-argument that
     ; that has a defense for the attack (it is not necessarily the case that this defense is
     ; already discovered)
+    ; the find-defense runs immediately for all agents working on a not fully researched
+    ; not-admissible argument, hence it is only once executed
     ifelse member? myargu not-admissible and [color] of myargu != red[
-      let myattacks []
-      ask myargu [
-        ask my-in-attacks [
-          ask myargu [
-            ask other-end [
-              if not any? in-attack-neighbors with [mytheory = [mytheory] of myargu][
-                set myattacks lput self myattacks
-              ]
-            ]
-          ]
-        ]
-      ]
-      if not empty? myattacks [
-        ask turtles with [breed = arguments and in-support-neighbor? myargu and color != gray][
-          let nextargu self
-          let moved false
-          foreach myattacks [
-            let myattacker self
-            if in-attack-neighbor? nextargu [
-              ask curagent [
-                if not moved [
-                  move-to nextargu
-                  set moved true
-                ]
-              ]
-            ]
-          ]
-        ]
+      
+      if not run-find-defense [
+        find-defense
+        set run-find-defense true
       ]
+      
     ][
+    
     ifelse any? turtles with [(breed = starts or breed = arguments) and
-      in-support-neighbor? myargu and color != gray and color != red and
+      in-support-neighbor? myargu and color != gray and color != red and color != turquoise and
       not (any? turtles with [breed = agents and xcor = [xcor] of myself and ycor = [ycor] of myself
         and member? self [collaborator-network] of curagent])] [
     let move-random random-float 1.0
+    
     ; every time step with 1/5 of the move-probability the agent moves
     ifelse move-random < (0.2 * move-probability * (1 - (color / 200))) [
       move-to one-of turtles with [(breed = starts or breed = arguments) and
-        in-support-neighbor? myargu and color != gray and color != red and not (
-          any? turtles with [breed = agents and xcor = [xcor] of myself and ycor = [ycor] of myself
-            and member? self [collaborator-network] of curagent])]
+        in-support-neighbor? myargu and color != gray and color != red and color != turquoise and 
+        not ( any? turtles with [breed = agents and xcor = [xcor] of myself and ycor = [ycor] of myself
+          and member? self [collaborator-network] of curagent])]
     ][
+    
     ; every 5th time step the agent mover with the full move-probability, that depends a bit on the color
-    if ticks != 0 and ticks mod 5 = 1 and move-random < move-probability * (1 - (color / 200)) [
+    if ticks != 0 and ticks mod 5 = 0 and move-random < move-probability * (1 - (color / 200)) [
```

Comment:
This code is part of the "move-around" procedure and determines the part where agents move with full move probability.  In this submit the "move around" procedure will only be performed by agents which are not communicating. Agents might be communicating in mod 5 = 0 rounds (=ticks).
The effect of this change is changing full movement probability movements from happening in mod 5 = 1 to mod 5 = 0 rounds. I.e. we're back to a situation where the majority of movement is being blocked by communicating. Afaik this is contrary to something we already agreed on some time ago: that movement should not be blocked by communicating. Is there any particular reason for reinstating the block?